src/virt_card.py

```diff
@@ -103,30 +92,55 @@ def run_cmd(self, cmd: str = None, expect: str = None, pin: bool = True,
                                           f"found in the output.")
                 shell.sendline(passwd)
 
-            if reject is not None:
-                out = shell.expect([reject, pexpect.EOF])
-                if out == 0:
-                    log.error("Disallowed pattern found")
-                    # add exception here
+        except PatternNotFound:
+            log.error(f"Command: {cmd}")
+            log.error(f"Output:\n{str(shell.before)}\n")
+            raise
+        return shell.read()
 
-            if expect is not None:
-                out = shell.expect([pexpect.TIMEOUT, expect], timeout=20)
-                if out != 1:
+    def check_output(self, output, expect: str = None,
+                zero_rc: bool = True, check_rc: bool = False,
+                reject: str = None):
+        """
+        Check text input for presence of expected and unexpected patterns.
+
+        Check for presence of expected (required) and disallowed patterns
+        in the text and raise exceptions if required pattern is missing or
+        if any of disallowed patterns is present.
+
+        Args:
+            expect: string or list of strings to be matched in the output
+            reject: string or list of string that cause failure if matched
+                    in the output
+            check_rc: inficates that return code of the cmd would be checked.
+                      If you put this parameter to False, but still want to
+                      check the return code of the cmd, use child.expect(["RC:0"])
+                      to check that return code of the cmd is 0.
+            zero_rc: indicates that it is expected from the command to end with
+                     non-zero exit code. Otherwise exception NonZeroReturnCode
+                     would be raised
+        """
+        if reject is not None:
+            if isinstance(reject, str):
+                reject = [reject]
+            for pattern in reject:
+                if pattern in output:
+                    raise DisallowedPatternFound(f"Disallowed pattern '{pattern}'"
```

Comment:
Put space between '{pattern}' and closing quiet, so it is more clear in the traceback log